tokio-util/src/codec/framed.rs

```diff
@@ -28,42 +28,97 @@ pin_project! {
     /// [`Sink`]: futures_sink::Sink
     /// [`AsyncRead`]: tokio::io::AsyncRead
     /// [`Decoder::framed`]: crate::codec::Decoder::framed()
-    pub struct Framed<T, U> {
+    pub struct Framed<T, U, State = RWFrames> {
```

Comment:
Is this a breaking change?

Reply:
It isn't, but it would allow code to be written that puts expects certain constraints to be allowed. I think [sealing it](https://rust-lang.github.io/api-guidelines/future-proofing.html) would be a good idea if we don't want this to be extensible

* https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md#minor-change-adding-defaulted-type-parameters